### ISSUE TITLE
Ability to specify absolute or relative path for assetFileNames

### ISSUE THREAD

**Issue by daleyjem** [rollup/rollup#3099]
It appears as though `assetFileNames` is relative to my specified `output.dir`.

If I use `preserveModules: true` with 2 different output formats at directories `./dist/es/` and `./dist/cjs/`... I would like to emit assets from my custom plugin to one single, shared directory. Instead, it emits the same asset files to 2 separate locations.

Is there something I'm missing for my particular case?

### Expected Behavior / Situation

I would like to use a relative path (i.e. `../assets/[name]-[hash].[ext]`)

### Actual Behavior / Situation

Absolute or relative paths aren't supported

### Modification Proposal

Ability to use absolute or relative path for `assetFileNames`

**Comment by lukastaegert** [comment]
I think I understand your use case but I am not sure this can be achieved easily and also slightly defeats the idea that the build produces a self-contained directory. What you *could* do is
* Use `output.dir: dist` for both formats
* Use naming patterns with format specific directories for chunks to avoid conflicts:
  * `output.entryFileNames: "[format]/[name].js"`
  * `output.chunkFileNames: "[format]/[name]-[hash].js"`
* Place your assets at the top level: `output.assetFileNames: "[name]-[hash].[ext]"`

Note though that it is your own responsibility to make sure the emitted assets are the same for each format as in this case, subsequent outputs will overwrite the assets of previous outputs.

**Comment by lukastaegert** [comment]
> Note though that it is your own responsibility to make sure the emitted assets are the same for each format as in this case, subsequent outputs will overwrite the assets of previous outputs

Thinking more about this, the `[hash]` will probably prevent this.

Might be interesting to document this approach somewhere on the website as I assume this is a common pattern also when creating ESM websites with e.g. a SystemJS fallback.

**Comment by DanielOrtel** [comment]
I don't agree with the idea that build should always produce one self-contained directory. While I do agree for JS files/types, where assets go should be configurable. So far I've relied on doing a `.dist` and a `.public` directory for my libraries, and it was cleaner. 

The output directory contains the type mappings as well, which creates a lot of noise. For example, I already have a `.dist/assets` folder, because my source code has an assets folder, so the typemapping creates one. And even changing the `assetFileNames` to `public/` doesn't help, because there's no reason another developer looking at 50 separate folders in the output directory realises that the `public` folder would actually contain the static assets, which, that developer needs to handle separately, since static assets are not automatically bundled when importing from a library. 

All in all, IMO, this is a needless, opinionated restriction, which at best solves some accidental outputs outside of the project directory. But that could be solved by restricting the pathing to either `subdirectory/pattern` or `./subdirectory/pattern`, where the latter would take the root of the project and the former would take the root of the output directory.
